### PR TITLE
enable integration tests for remote-client

### PR DESCRIPTION
**Pull request by baude**
first pass at enabling a swath of integration tests for the
remote-client.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: test/e2e/info_test.go [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
-
 package integration
 
 import (
-	"os"
-
 	. "github.com/containers/libpod/test/utils"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	"os"
```
Comment (mheon) [inline test/e2e/info_test.go]: Ordering?

**Comment by jwhonce** [comment]
LGTM

**Comment by baude** [comment]
@giuseppe any theories on the failed rootless tests?

**Comment by baude** [comment]
Needs https://github.com/varlink/go/pull/6

File: API.md [hunk 2]
```diff
@@ -259,6 +259,10 @@ in the [API.md](https://github.com/containers/libpod/blob/master/API.md) file in
 
 [error ContainerNotFound](#ContainerNotFound)
 
+[error ErrCtrStateInvalid](#ErrCtrStateInvalid)
```
Comment (mheon) [inline API.md]: Wait I added this error already

Comment (mheon) [reply API.md]: error InvalidState (id: string, reason: string)

File: cmd/podman/rm.go [hunk 3]
```diff
@@ -2,7 +2,6 @@ package main
 
 import (
 	"fmt"
-
```
Comment (mheon) [inline cmd/podman/rm.go]: This shouldn't be removed

File: libpod/runtime.go [hunk 4]
```diff
@@ -922,7 +922,7 @@ func makeRuntime(ctx context.Context, runtime *Runtime) (err error) {
 		if os.IsNotExist(errors.Cause(err)) {
 			manager, err = lock.NewSHMLockManager(lockPath, runtime.config.NumLocks)
 			if err != nil {
-				return err
+				return errors.Wrapf(err, "failed to get new shm lock")
```
Comment (mheon) [inline libpod/runtime.go]: nit: change to "failed to get new shm lock manager" - less confusing

File: pkg/adapter/containers_remote.go [hunk 5]
```diff
@@ -234,15 +234,24 @@ func (r *LocalRuntime) StopContainers(ctx context.Context, cli *cliconfig.StopVa
...
 	if err != nil {
-		return ok, failures, err
+		return ok, failures, TranslateError(err)
 	}
 
 	for _, id := range ids {
-		stopped, err := iopodman.StopContainer().Call(r.Conn, id, int64(cli.Timeout))
-		if err != nil {
+		if _, err := iopodman.StopContainer().Call(r.Conn, id, int64(cli.Timeout)); err != nil {
+			if errors.Cause(TranslateError(err)) == libpod.ErrCtrStopped {
```
Comment (mheon) [inline pkg/adapter/containers_remote.go]: Worth calling TranslateError only once and working with the translated error?

File: test/e2e/libpod_suite_remoteclient_test.go [hunk 6]
```diff
@@ -50,34 +54,95 @@ func PodmanTestCreate(tempDir string) *PodmanTestIntegration {
...
+		fmt.Println(i.Pid())
+		fmt.Println("++++++++++++++++++++++++")
+
+	} else {
+		p.ResetVarlinkAddress()
+		//pgid, err := syscall.Getpgid(p.VarlinkSession.Pid)
+		//if err != nil {
+		//	fmt.Printf("Unable to get pgid\n")
+		//}
+		//fmt.Printf("kill %d\n", pgid)
```
Comment (mheon) [inline test/e2e/libpod_suite_remoteclient_test.go]: Can we remove the commented code?

File: test/e2e/libpod_suite_remoteclient_test.go [hunk 7]
```diff
@@ -50,34 +54,95 @@ func PodmanTestCreate(tempDir string) *PodmanTestIntegration {
...
-		if err := os.Remove(socket); err != nil {
-			fmt.Println(err)
+		if err := varlinkSession.Kill(); err != nil {
+			fmt.Fprintf(os.Stderr, "error on varlink stop-kill %q", err)
+		}
+		i, err := varlinkSession.Wait()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "error on varlink stop-wait %q", err)
+		}
+		fmt.Println("++++++++++++++++++++++++")
```
Comment (mheon) [inline test/e2e/libpod_suite_remoteclient_test.go]: Debugging?

File: API.md (hunk 2 above)
Comment (baude) [reply API.md]: not for varlink

**Comment by baude** [comment]
fixed up

**Pull request merged**